pkg/operator/ceph/csi/secrets.go

```diff
@@ -183,38 +237,199 @@ func CreateCSISecrets(context *clusterd.Context, clusterInfo *client.ClusterInfo
 	if err != nil {
 		return errors.Wrap(err, "failed to create csi rbd provisioner ceph keyring")
 	}
 
 	// Create CSI RBD Node Ceph key
-	csiRBDNodeSecretKey, err := createCSIKeyringRBDNode(k)
+	rbdNodeName, rbdNodeKey, rbdNodeKeyCount, _, err := createCSIKeyring(context, clusterInfo, cephCluster, k, csiKeyringRBDNodeUsername, cephCSIKeyringRBDNodeCaps())
 	if err != nil {
 		return errors.Wrap(err, "failed to create csi rbd node ceph keyring")
 	}
 
 	// Create CSI Cephfs provisioner Ceph key
-	csiCephFSProvisionerSecretKey, err := createCSIKeyringCephFSProvisioner(k)
+	cFSProvName, cFSProvKey, cFSProvKeyCount, _, err := createCSIKeyring(context, clusterInfo, cephCluster, k, csiKeyringCephFSProvisionerUsername, cephCSIKeyringCephFSProvisionerCaps())
 	if err != nil {
 		return errors.Wrap(err, "failed to create csi cephfs provisioner ceph keyring")
 	}
 
 	// Create CSI Cephfs node Ceph key
-	csiCephFSNodeSecretKey, err := createCSIKeyringCephFSNode(k)
+	cFSNodeName, cFSNodeKey, cFSNodeKeyCount, shouldRotate, err := createCSIKeyring(context, clusterInfo, cephCluster, k, csiKeyringCephFSNodeUsername, cephCSIKeyringCephFSNodeCaps())
 	if err != nil {
 		return errors.Wrap(err, "failed to create csi cephfs node ceph keyring")
 	}
 
+	csiSecretContent := csiSecretStore{
+		CsiRBDProvisionerSecret:    {Name: rbdProvName, Key: rbdProvKey},
+		CsiRBDNodeSecret:           {Name: rbdNodeName, Key: rbdNodeKey},
+		CsiCephFSProvisionerSecret: {Name: cFSProvName, Key: cFSProvKey},
+		CsiCephFSNodeSecret:        {Name: cFSNodeName, Key: cFSNodeKey},
+	}
+
 	// Create or update Kubernetes CSI secret
-	if err := createOrUpdateCSISecret(clusterInfo, csiRBDProvisionerSecretKey, csiRBDNodeSecretKey, csiCephFSProvisionerSecretKey, csiCephFSNodeSecretKey, k); err != nil {
+	if err := createOrUpdateCSISecret(clusterInfo, csiSecretContent, k); err != nil {
 		return errors.Wrap(err, "failed to create kubernetes csi secret")
 	}
 
+	currentKeyCount := max(rbdProvKeyCount, rbdNodeKeyCount, cFSProvKeyCount, cFSNodeKeyCount)
+	err = updateCephStatusWithCephxStatus(context, clusterInfo, cephCluster, clusterNamespaced, shouldRotate, currentKeyCount)
+	if err != nil {
+		return err
+	}
+
 	return nil
 }
 
+// updateCephStatusWithCephxStatus updates the cephCluster status with CSI cephxStatus
+func updateCephStatusWithCephxStatus(context *clusterd.Context, clusterInfo *client.ClusterInfo, cephCluster *cephv1.CephCluster,
+	namespacedName types.NamespacedName, shouldRotate bool, currentKeyCount int,
+) error {
+	logger.Info("updating cephStatus with CSI cephxStatus")
+	cephxStatus := keyring.UpdatedCephxStatus(shouldRotate, cephCluster.Spec.Security.CephX.Daemon, clusterInfo.CephVersion, *cephCluster.Status.Cephx.CSI.CephxStatus)
+
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		cephCluster.Status.Cephx.CSI.CephxStatus = &cephxStatus
+		cephCluster.Status.Cephx.CSI.PriorKeyCount = uint8(currentKeyCount) //nolint:gosec // disable G115
+
```

Comment:
I realize the confusion I added here after my last review. Sorry. As you have written the code here, it is correct to use the "cached" version of `cephCluster` passed in to check `UpdatedCephxStatus()`. The cached version of the config is what was used to handle the CSI rotation logic prior to this point, so we want that for consistency. 

However, in the update loop, we do need to `Get()` the most recent CephCluster each time in order to ensure this can recover from conflicts. This will ensure that we are applying the update using the latest object resourceVersion, which is the mechanism by which kubernetes detects conflicts

Reply:
updated the logic